homework/62374/java2html/Lexer.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include "CommonTypes.h"
+
+enum class TokenType
+{
+	// Single-character tokens.
+	LeftParen,
+	RightParen,
+	LeftBrace,
+	RightBrace,
+	Comma,
+	Dot,
+	Minus,
+	Plus,
+	Semicolon,
+	Star,
+	Division,
+	Modulo,
+	BitwiseNot,
+	BitwiseAnd,
+	BitwiseXor,
+	BitwiseOr,
+	QuestionMark,
+	Colon,
+	LeftSquareBracket,
+	RightSquareBracket,
+
+	// One or two character tokens
+	Bang,
+	BangEqual,
+	Equal,
+	EqualEqual,
+	StrictEqual,
+	StrictNotEqual,
+	Greater,
+	GreaterEqual,
+	Less,
+	LessEqual,
+	MinusMinus,
+	PlusPlus,
+	LeftShift,
+	RightShift,
+	LogicalAnd,
+	LogicalOr,
+	StarEqual,
+	DivideEqual,
+	ModuloEqual,
+	PlusEqual,
+	MinusEqual,
+	LeftShiftEqual,
+	RightShiftEqual,
+	BitwiseAndEqual,
+	BitwiseXorEqual,
+	BitwiseOrEqual,
+	Backslash,
+
+	Identifier,
+	String,
+	Number,
+
+	// Keywords. ECMAScript 2015
+	Abstract,
+	Package,
+	Synchronize,
+	Boolean,
```

Comment:
It would have been better to extend the JSImpl to allow different languages, than duplicating the JavaScript implementation